lib/internal/policy/sri.js

```diff
@@ -17,22 +20,22 @@ const kHASH_ALGO = 'sha(?:256|384|512)';
 // Base64
 const kHASH_VALUE = '[A-Za-z0-9+/]+[=]{0,2}';
 const kHASH_EXPRESSION = `(${kHASH_ALGO})-(${kHASH_VALUE})`;
-const kOPTION_EXPRESSION = `(${kVCHAR}*)`;
+// Ungrouped since unused
+const kOPTION_EXPRESSION = `(?:${kVCHAR}*)`;
 const kHASH_WITH_OPTIONS = `${kHASH_EXPRESSION}(?:[?](${kOPTION_EXPRESSION}))?`;
 const kSRIPattern = RegExp(`(${kWSP}*)(?:${kHASH_WITH_OPTIONS})`, 'g');
-const { freeze } = Object;
-Object.seal(kSRIPattern);
-const kAllWSP = RegExp(`^${kWSP}*$`);
-Object.seal(kAllWSP);
+freeze(kSRIPattern);
 
 const BufferFrom = require('buffer').Buffer.from;
+const RealArrayPrototype = getPrototypeOf([]);
 
 const parse = (str) => {
-  kSRIPattern.lastIndex = 0;
   let prevIndex = 0;
-  let match;
-  const entries = [];
-  while (match = RegExpPrototype.exec(kSRIPattern, str)) {
+  const entries = setPrototypeOf([], null);
+  for (const match of StringPrototype.matchAll(
```

Comment:
nit: would be more readable if we save the iterator instead of squeezing the whole thing into the for statement.

Reply:
should be fixed